atlas-db-controller/controller.go

```diff
@@ -244,6 +272,81 @@ func (c *Controller) processNextWorkItem(q workqueue.RateLimitingInterface, sync
 	return true
 }
 
+func (c *Controller) syncSchema(key string) error {
+	glog.Infof("Schema key: %v", key)
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return err
+	}
+
+	// Get the Schema resource with this namespace/name
+	schema, err := c.schemasLister.DatabaseSchemas(namespace).Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("schema '%s' in work queue no longer exists", key))
+			return nil
+		}
+		return err
+	}
+	glog.Info(schema.Spec)
+	dbName := schema.Spec.Database
+	db, err := c.dbsLister.Databases(namespace).Get(dbName)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("cannot get database `%s`: %s", dbName, err))
+		return err
+	}
+
+	glog.Info(db.Spec.ServerType)
+	glog.Info(db.Spec.Dsn)
+
+	if db.Spec.ServerType != "postgres" { //  && db.Spec.ServerType != ...
+		err = fmt.Errorf("unsupported database server type `%s`", db.Spec.ServerType)
+		runtime.HandleError(err)
+		return err
+	}
+
+	mgrt, err := migrate.New(schema.Spec.Git, db.Spec.Dsn)
+	if err != nil {
+		err = fmt.Errorf("cannot create migrate: %s", err)
+		runtime.HandleError(err)
+		return err
+	}
+	defer mgrt.Close()
+
+	ver, dirt, err := mgrt.Version()
```

Comment:
Does this create the migration table in an empty DB? If not, where is that done?

Reply:
This is up to db driver implementation. postgres driver creates the table during Open(), i.e. inside of `mgrt, err := migrate.New(...)`. The mysql, cassandra, sqlite do this the same way.